### ISSUE TITLE
* Acc@1 0.092 (99.908) Acc@5 0.338 (99.662)

### ISSUE THREAD

**Issue by xuzhiyuan1022** [Adamdad/kat#33]
/home/anaconda3/bin/conda run -n kat --no-capture-output python /home/kat/validate.py /home/imagenet/ --model kat_tiny_patch16_224 --checkpoint kat_tiny_patch16_224_1f3ad3b2e69821f3d412f2924cf159a0e266f142d739cb68f68f796f5a0fe289.pth -b 512 
/home/rational_kat_cu/kat_rational/kat_1dgroup.py:10: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @torch.cuda.amp.custom_fwd(cast_inputs=torch.float32)
/home/rational_kat_cu/kat_rational/kat_1dgroup.py:32: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(ctx, grad_output):
Validating in float32. AMP not enabled.
Loaded state_dict from checkpoint 'kat_tiny_patch16_224_1f3ad3b2e69821f3d412f2924cf159a0e266f142d739cb68f68f796f5a0fe289.pth'
Model kat_tiny_patch16_224 created, param count: 5718328
Data processing configuration for current model + dataset:
	input_size: (3, 224, 224)
	interpolation: bicubic
	mean: (0.485, 0.456, 0.406)
	std: (0.229, 0.224, 0.225)
	crop_pct: 0.875
	crop_mode: center
Test: [   0/98]  Time: 1.702s (1.702s,  300.89/s)  Loss:  9.5117 (9.5117)  Acc@1:   0.000 (  0.000)  Acc@5:   0.586 (  0.586)
Test: [  10/98]  Time: 0.099s (0.414s, 1235.78/s)  Loss:  9.5943 (9.5137)  Acc@1:   0.000 (  0.036)  Acc@5:   0.000 (  0.231)
Test: [  20/98]  Time: 0.102s (0.403s, 1271.90/s)  Loss:  9.5652 (9.5066)  Acc@1:   0.000 (  0.084)  Acc@5:   0.000 (  0.279)
Test: [  30/98]  Time: 0.100s (0.368s, 1390.58/s)  Loss:  9.5710 (9.5108)  Acc@1:   0.000 (  0.082)  Acc@5:   0.391 (  0.290)
Test: [  40/98]  Time: 0.100s (0.373s, 1372.38/s)  Loss:  9.5144 (9.5206)  Acc@1:   0.000 (  0.091)  Acc@5:   0.977 (  0.305)
Test: [  50/98]  Time: 0.101s (0.358s, 1431.60/s)  Loss:  9.4829 (9.5188)  Acc@1:   0.195 (  0.096)  Acc@5:   0.391 (  0.310)
Test: [  60/98]  Time: 0.132s (0.362s, 1414.92/s)  Loss:  9.4237 (9.5226)  Acc@1:   0.391 (  0.096)  Acc@5:   0.977 (  0.320)
Test: [  70/98]  Time: 0.144s (0.355s, 1441.42/s)  Loss:  9.5024 (9.5218)  Acc@1:   0.000 (  0.088)  Acc@5:   0.195 (  0.319)
Test: [  80/98]  Time: 0.377s (0.361s, 1416.44/s)  Loss:  9.5698 (9.5249)  Acc@1:   0.000 (  0.094)  Acc@5:   0.586 (  0.333)
Test: [  90/98]  Time: 0.103s (0.355s, 1443.01/s)  Loss:  9.4263 (9.5242)  Acc@1:   0.391 (  0.094)  Acc@5:   0.781 (  0.335)
 * Acc@1 0.092 (99.908) Acc@5 0.338 (99.662)
--result
{
    "model": "kat_tiny_patch16_224",
    "top1": 0.092,
    "top1_err": 99.908,
    "top5": 0.338,
    "top5_err": 99.662,
    "param_count": 5.72,
    "img_size": 224,
    "crop_pct": 0.875,
    "interpolation": "bicubic"
}
thank for your sharing work. I already downloaded the checkpoint“'kat_tiny_patch16_224_1f3ad3b2e69821f3d412f2924cf159a0e266f142d739cb68f68f796f5a0fe289.pth'”，but when i run validate.py  with param" /home/imagenet/ --model kat_tiny_patch16_224 --checkpoint kat_tiny_patch16_224_1f3ad3b2e69821f3d412f2924cf159a0e266f142d739cb68f68f796f5a0fe289.pth -b 512 "
the result is very terrible

**Comment by Adamdad** [comment]
Hello @xuzhiyuan1022 , May I ask which version of ImageNet you're using? Different versions can have different class IDs—for example, a dog might be class 1 in one version but class 100 in another. This mismatch could be the reason for the extremely low results you’re seeing. You might find this repo helpful as a reference for the correct version I used: https://github.com/Jasonlee1995/ImageNet-1K.

Hope this helps!

**Comment by xuzhiyuan1022** [comment]
> Hello [@xuzhiyuan1022](https://github.com/xuzhiyuan1022) , May I ask which version of ImageNet you're using? Different versions can have different class IDs—for example, a dog might be class 1 in one version but class 100 in another. This mismatch could be the reason for the extremely low results you’re seeing. You might find this repo helpful as a reference for the correct version I used: https://github.com/Jasonlee1995/ImageNet-1K.
> 
> Hope this helps!
Thank you for your reply, I replaced the dataset you provided and retested，it is working !
